spectator-web-spring/src/main/java/com/netflix/spectator/controllers/MetricsController.java

```diff
@@ -0,0 +1,229 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.controllers;
+
+import com.netflix.spectator.api.Counter;
+import com.netflix.spectator.api.DistributionSummary;
+import com.netflix.spectator.api.Gauge;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.LongTaskTimer;
+import com.netflix.spectator.api.Measurement;
+import com.netflix.spectator.api.Meter;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Timer;
+
+import com.netflix.spectator.controllers.filter.ChainedMeasurementFilter;
+import com.netflix.spectator.controllers.filter.MeasurementFilter;
+import com.netflix.spectator.controllers.filter.PrototypeMeasurementFilter;
+import com.netflix.spectator.controllers.filter.TagMeasurementFilter;
+import com.netflix.spectator.controllers.model.ApplicationRegistry;
+import com.netflix.spectator.controllers.model.MetricValues;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+
+/**
+ * Provides an HTTP endpoint for polling spectator metrics.
+ */
+@RequestMapping("/spectator/metrics")
+@RestController
+@ConditionalOnExpression("${spectator.endpoint.enabled:false}")
+public class MetricsController {
+  @Autowired
+  private Registry registry;
+
+  @Value("${spectator.applicationName:}")
+  private String applicationName;
+
+  @Value("${spectator.applicationVersion:}")
+  private String applicationVersion;
+
+  @Value("${spectator.endpoint.tagFilter.meterNameRegex:.*}")
+  private String meterNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagNameRegex:}")
+  private String tagNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagValueRegex:}")
+  private String tagValueRegex;
+
+  @Value("${spectator.endpoint.prototypeFilter.path:}")
+  private String prototypeFilterPath;
+
+  private MeasurementFilter defaultMeasurementFilter = null;
+
+  /**
+   * The default measurement filter is configured through properties.
+   */
+  public MeasurementFilter getDefaultMeasurementFilter() throws IOException {
```

Comment:
It looks like this is only called as part of a GET request handler, two questions:

1. Can the visibility be restricted?
2. Are there any thread safety concerns here?  I noticed it mentioned for other classes they were expected to be executed from a single thread, but wanted to ask here because I'm assuming the request handler could be called concurrently.